tests/dialects/test_duckdb.py

```diff
@@ -968,6 +964,32 @@ def test_cast(self):
         self.validate_identity("CAST(x AS BINARY)", "CAST(x AS BLOB)")
         self.validate_identity("CAST(x AS VARBINARY)", "CAST(x AS BLOB)")
         self.validate_identity("CAST(x AS LOGICAL)", "CAST(x AS BOOLEAN)")
+        self.validate_identity(
+            "CAST(ROW(1, ROW(1)) AS STRUCT(number BIGINT, row STRUCT(number BIGINT)))"
+        )
+        self.validate_identity(
+            "123::CHARACTER VARYING",
+            "CAST(123 AS TEXT)",
+        )
+        self.validate_identity(
+            "CAST([[STRUCT_PACK(a := 1)]] AS STRUCT(a BIGINT)[][])",
+            "CAST([[{'a': 1}]] AS STRUCT(a BIGINT)[][])",
+        )
+        self.validate_identity(
+            "CAST([STRUCT_PACK(a := 1)] AS STRUCT(a BIGINT)[])",
+            "CAST([{'a': 1}] AS STRUCT(a BIGINT)[])",
+        )
+
+        self.validate_all(
+            "CAST(x AS DECIMAL(38, 0))",
+            read={
+                "snowflake": "CAST(x AS NUMBER)",
```

Comment:
should we maintain identity for snowflake? so if it's a decimal(38, 0) we write number

Reply:
Not sure, I also considered this but then we'd change an explicit `DECIMAL(38, 0)` into the implicit `NUMBER` type, so we don't attain 100% identity, so I didn't feel strongly about adding complexity to preserve `NUMBER`.

I'm happy with either, let me know what you prefer and if needed I'll refactor.